ERCS/erc-7779.md

```diff
@@ -0,0 +1,191 @@
+
+7702 Accounts do plan to use a custom non-zero storage slot to avoid storage collision as much as possible, however, there hasn’t been a standardized approach on how to fetch them.
+
+This function provides a standardized approach for wallets and other applications to check the base storage slots of an account, and verify if the base storage slots are far enough from the newly to-be-redelegated account’s base storage slot.
+
+Note that there could be some exceptions for mappings, etc depending on how account manages storage.
+
+This provides Wallets and applications a standard approach to fetch the base storage slots for verification rather than just relying on the probability of hash.
+
+If the account uses external storage, it should return ERC Number prefixed slot with the external storage contract address concatenated.
+
+```solidity
+<ERC-Number><ERC-Number><ERC-Number>..N...<Storage-Contract-Address>
+```
+For example, if the storage contract address is `0xAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA` the returned value should be
+```
+0x777977797779777977797779AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
+```
+
+When the account is delegated(e.g., during the account initialization stage), the account implementation should append the base storage slot of the account to the following slot position: `keccak(keccak('InteroperableDelegatedAccount.ERC.Storage')-1) & ~0xff` .
+
+The storage variable would be using the `bytes32[]` type and is an append-only variable which newly delegated accounts append its storage slot hash. The account may choose to not append its storage base to an array if there is an identical entry that already exist.
+
+In case the account identifies that there are colliding storage bases, the account can perform further storage verification either off-chain or on-chain and decide whether the delegation would happen.
+
+Wallet may revert the delegation if the colliding storage includes a suspicious storage values that may target the user(e.g., shadow signer, etc).
+
+### `onRedelegation()`
+
+This function is to prepare for the redelegation to a new account.
+
+When this function is called, the existing delegated account should perform actions to not limit or impact the user when the user redelegates to a new account as much as possible.
+
+For example, the account could uninitialize the storage variables as much as it can to provide clean storage for new wallet.
+
+This standard, however, does not explicitly state the behavior to be done during this function call as wallet implementations have very distinctive architecture and details.
+
+**The standard expect the wallet implementation to revert it back to a clean storage state ***as much as possible*** with this function.**
+
+`onRedelegation()` should validate if the caller is indeed the authorized user by checking the `msg.sender` value.
+
+This could also be done through a `self-call` if a custom validation scheme is implemented or at the wallet's discretion as a side case.
```

Comment:
```suggestion
This could also be done through a "self-call" if a custom validation scheme is implemented or at the wallet's discretion as a side case.
```